### PR TITLE
Added TextChannel#deleteMessages()

### PR DESCRIPTION
**Pull request by Frederikam**
- Added TextChannel#deleteMessages() https://github.com/hammerandchisel/discord-api-docs/issues/40
- Added project files specific to Netbeans (harmless)
- Fixed a typo because I am a nazi

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/entities/TextChannel.java [hunk 1]
```diff
@@ -34,4 +36,14 @@
      *      The String needed to mention this Channel
      */
     String getAsMention();
+    
+    /**
+     * Bulk deletes a list of messages. Must not be more than 100 messages at a time.
+     * You must have permission to delete a message. You may always delete your own messages. In the event that you do not have permission to delete a message, the message will be skipped.
```
Comment (DinosParkour) [inline src/main/java/net/dv8tion/jda/entities/TextChannel.java]: This isn't quite correct, you need MESSAGE_MANAGE to bulk delete anyway so all messages in the list will be deleted..

File: src/main/java/net/dv8tion/jda/entities/impl/TextChannelImpl.java [hunk 2]
```diff
@@ -355,6 +355,38 @@ public TextChannelImpl setPosition(int position)
...
+    @Override
+    public void deleteMessages(List<Message> messages)
+    {
+        if(messages.size() > 100)
+        {
+            throw new IllegalArgumentException("Can't delete more than 100 messages at a time, got " + messages.size());
+        }
+        else if(messages.size() == 1)
+        {
+            messages.get(0).deleteMessage();
```
Comment (DinosParkour) [inline src/main/java/net/dv8tion/jda/entities/impl/TextChannelImpl.java]: Check if the Bot has the permission required to delete the message (aka if it's not its own msg)?

File: src/main/java/net/dv8tion/jda/entities/impl/TextChannelImpl.java [hunk 3]
```diff
@@ -355,6 +355,38 @@ public TextChannelImpl setPosition(int position)
...
+            return;
+        }
+        else if(messages.isEmpty())
+        {
+            return;
+        }
+        
+        if(PermissionUtil.checkPermission(getJDA().getSelfInfo(), Permission.MESSAGE_MANAGE, this) == false)
+        {
+            throw new PermissionException(Permission.MANAGE_PERMISSIONS, "Must have MESSAGE_MANAGE in order to bulk delete messages in this channel regardless of author.");
```
Comment (DinosParkour) [inline src/main/java/net/dv8tion/jda/entities/impl/TextChannelImpl.java]: Did you mean to throw
```java
new PermissionException(Permission.MESSAGE_MANAGE, ...
```

File: src/main/java/net/dv8tion/jda/entities/impl/TextChannelImpl.java (hunk 2 above)
Comment (Frederikam) [inline src/main/java/net/dv8tion/jda/entities/impl/TextChannelImpl.java]: `messages.get(0).deleteMessage();` will throw an exception in such a case.

**Comment by kantenkugel** [comment]
looks good!
Waiting for DV8 before merging tho

**Comment by kantenkugel** [comment]
you could (if you want) properly link the deleteMessage method ( `{@link net.dv8tion.jda.entities.Message#deleteMessage}`)

**Comment by Frederikam** [comment]
Thanks Kant. Went ahead and committed that.

**Comment by DV8FromTheWorld** [comment]
Code looks good. Please compress the commits into a single commit considering many are for simple changes.
If possible, cleanse the first commit which is a merge and unneeded. If you can't, that's fine. Just compress and drop a message when done and I'll merge it

**Comment by Frederikam** [comment]
DV8 you can just squash all the commits into one when merging this PR. This will produce a single commit for this entire PR.

**Comment by DV8FromTheWorld** [comment]
Actually, I missed something: Line 376 of TextChannelImpl, please change the if statement to use `!` instead of `== false`

**Comment by DV8FromTheWorld** [comment]
And add a newline at the end of the .gitignore file

**Comment by DV8FromTheWorld** [comment]
Okay, I didn't know github had that functionality now. Dope. Other changes still stand.

**Comment by Frederikam** [comment]
:+1:

**Comment by DV8FromTheWorld** [comment]
Looks good. Question: what happens if you pass ids for messages sent in a different channel? What about if you pass it messages built with the message builder? We might want to add catches for those situations.

Otherwise, this is mergable.

**Comment by Frederikam** [comment]
The endpoint doesn't care if the message belongs in another channel or if it doesn't exist. However, if you use an unsent message from a builder the message array will contain a `""` which is invalid formatting, causing a silent fail. Will fix.

**Comment by Frederikam** [comment]
Unsent messages will now be ignored

**Comment by kantenkugel** [comment]
You ready to merge @Frederikam ? BTW: i find it funny how you use the `lastProcessedMessage` var instead of just using `ids.get(0)` xD

**Comment by Frederikam** [comment]
Ready to merge.

`lastProcessedMessage` exists because `ids.get(0)` is only the id (String) of the message. It seems dirty to convert the id to a Message object or just do a delete directly using REST.

**Comment by kantenkugel** [comment]
oh right... kk

**Comment by DV8FromTheWorld** [comment]
Im concerned that silently removing and returning when dealing with messages built from MessageBuilder may prove to cause hidden problems for some users not knowing why their messages aren't being deleted because that are attempting to delete a MessageBuilder built message, however, I'll leave it for now.

**Comment by Frederikam** [comment]
I don't see the problem. The desired result is to not have the message exist in the given channel. If the message is never sent this goal is already fulfilled.

**Pull request merged**